.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,31 @@
+name: "Build on PR"
```

Comment:
Nit: this is more like "verify" or "test" on PR. 

My personal preference (but this is 100% a matter of taste, so I'm not being presecriptive here) is to align the naming with the purpose (testing/verification) rather than the context (pull request), but might just be me.